### DISCUSSION TITLE
How to unmap lsp Hover (K)

### DISCUSSION THREAD

**Discussion by StitiFatah** [LazyVim/LazyVim#758] [Q&A] [answered]
Hi, 

I'd want to map Shift+i to hover and Shift+k to next buffer because of ... muscle memory,  for shift+I the following does the job :

```lua
vim.api.nvim_create_autocmd("LspAttach", {
  callback = function(args)
    vim.keymap.set("n", "I", vim.lsp.buf.hover, { buffer = args.buf })
  end,
})
```
but I'm not able to unmap Shift+k with 

```lua
vim.keymap.del("n", "K")
-- 
vim.keymap.del("n", "<S-k>")
```

Or maybe there is a way to overwrite it with `vim.keymap.set("n", "<S-k>", "<cmd>bn<CR>")` ? 

I've searched the docs and haven't foud anything mentioning hover besides the Keymaps cheatsheet. 

Thanks in advance.

**Comment by dpetka2001** [comment, marked as answer]
It's in the [docs](https://www.lazyvim.org/plugins/lsp#%EF%B8%8F-customizing-lsp-keymaps) although changing LSP keymaps is a bit different procedure than normal keymapping.

**Reply by StitiFatah** [reply]
Thanks it works fine, I assumed the keys were in the "keys" table like for other plugins but they're here https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/lsp/keymaps.lua

**Comment by Gandharikos** [comment]
```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      inlay_hints = { enabled = true },
    },
    init = function()
      local lsp_keys = require("lazyvim.plugins.lsp.keymaps")
      local keys = lsp_keys.get()
      for _, key in ipairs(keys) do
        if key[1] == "K" then
          key[1] = "I"
        elseif key[1] == "gK" then
          key[1] = "gi"
        elseif key[1] == "<c-k>" then
          key[1] = "<c-h>"
        end
      end
    end,
  },
```
